### PR TITLE
Added prims [C]

### LINKED ISSUE #184: Prims [C]
Link for algo- http://www.geeksforgeeks.org/greedy-algorithms-set-5-prims-minimum-spanning-tree-mst-2/

### PR DESCRIPTION
**Pull request by hims1911**
Fixes #184 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous timespan of 7 days.

### CODE REVIEW

File: prims/prims.c [hunk 1]
```diff
@@ -0,0 +1,48 @@
+#include <stdio.h>
+#include <string.h>
+
+# define SIZE 5
+# define INFINITY 1000
+
+// This function finds the minimal spanning tree by Prim's Algorithm
+void prims(int G[SIZE][SIZE], int *parent) {
+    int select[SIZE], i, j, k;
+    int min_dist = INFINITY;
```
Comment (Monal5031) [inline prims/prims.c]: To solve this issue remove this line and on line 16 make it `int min_dist = INFINITY`

File: prims/prims.c [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    int	v1 = 0, v2 = 0;
```
Comment (Monal5031) [inline prims/prims.c]: Replace the tabspace here with a single whitespace

Comment (Monal5031) [inline prims/prims.c]: Replace the tabspace here with a single whitespace

File: prims/prims.c [hunk 3]
```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <string.h>
+
+# define SIZE 5
+# define INFINITY 9999
+
+// This function finds the minimal spanning tree by Prim's Algorithm
+void prims(int G[SIZE][SIZE], int *parent) {
+    int select[SIZE], i, j, k;
+    int	v1 = 0, v2 = 0;
```
Comment (Monal5031) [inline prims/prims.c]: Just fix this tabspace after `int`, rest of the things looks good.

File: prims/prims.c [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    for (k = 1; k < SIZE; ++k) {
+        int min_dist = INFINITY;
+        for (i = 0; i < SIZE; ++i) {  // Select an edge such that one vertex is selected and other is not and the edge
+            for (j = 0; j < SIZE; ++j) {  // has the least weight.
+                if (G[i][j] && ((select[i] && !select[j]) || (!select[i] && select[j]))) {
+                    if (G[i][j] < min_dist) {  //obtained edge with minimum wt 
+                        min_dist = G[i][j];
+                        v1 = i;
+                        parent[j] = v1;
+				        v2 = j;   //picking up those vertices
```
Comment (Monal5031) [inline prims/prims.c]: Unnecessary tab's here.

Comment (hims1911) [inline prims/prims.c]: ya 👍

File: prims/prims.c (hunk 3 above)
Comment (hims1911) [inline prims/prims.c]: thnx 😄

**Comment by aviaryan** [comment]
@mohitkyadav We can't see any comments from your side.

File: prims/prims.c [hunk 5]
```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <string.h>
+
+# define SIZE 5
+# define INFINITY 9999
```
Comment (mohitkyadav) [inline prims/prims.c]: remove the space between `#` and `define`

File: prims/prims.c [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+                         {0, 5, 7, 9, 0},
+                        };
+    int i, j;
+    int parent[SIZE];
+    memset(parent, 0, SIZE);
+    printf("Edge\tWeight\n");
+    prims(G,parent);
+    for (i = 1; i < SIZE; ++i)
+        printf("%d - %d\t%d \n", parent[i], i, G[i][parent[i]]);
+	return 0;
```
Comment (mohitkyadav) [inline prims/prims.c]: Indentation going wrong here.

File: prims/prims.c [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+                    if (G[i][j] < min_dist) {  //obtained edge with minimum wt 
+                        min_dist = G[i][j];
+                        v1 = i;
+                        parent[j] = v1;
+                        v2 = j;  //picking up those vertices
+                    }
+                }
+            }
+        }
+        select[v1] = select[v2] = 1;
```
Comment (mohitkyadav) [inline prims/prims.c]: Remove space from the end of the line.

File: prims/prims.c [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+                    if (G[i][j] < min_dist) {  //obtained edge with minimum wt 
+                        min_dist = G[i][j];
+                        v1 = i;
+                        parent[j] = v1;
+                        v2 = j;  //picking up those vertices
+                    }
+                }
+            }
+        }
+        select[v1] = select[v2] = 0;
```
Comment (mohitkyadav) [inline prims/prims.c]: If we do ` select[v1] = select[v2] = 0;` your code throws RE(runtime error), [click here](http://ideone.com/BzBGDD)
And remove extra whitespace at the end of this line(28).

Comment (hims1911) [inline prims/prims.c]: yup 😄

**Comment by Monal5031** [comment]
@singhpratyush @aviaryan this one seems to be good to merge.

File: prims/prims.c [hunk 9]
```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <string.h>
+
+#define SIZE 5
+#define INFINITY 9999
```
Comment (singhpratyush) [inline prims/prims.c]: Better to use `int`'s max value.

File: prims/prims.c [hunk 10]
```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <string.h>
+
+#define SIZE 5
+#define INFINITY 32767
```
Comment (singhpratyush) [inline prims/prims.c]: What I suggested was that instead of `INFINITY`, you could use `INT_MAX` in the code.

Comment (hims1911) [inline prims/prims.c]: sorry -.-, i thought that infinity

**Comment by hims1911** [comment]
@singhpratyush i think all the changes are done...:D

**Pull request merged**